packages/components/src/Combobox/components/ComboboxContent/ComboboxContent.tsx

```diff
@@ -11,8 +11,13 @@ import { useComboboxAccessibility } from "../../hooks/useComboboxAccessibility";
 import { ComboboxContentProps, ComboboxOption } from "../../Combobox.types";
 
 export function ComboboxContent(props: ComboboxContentProps): JSX.Element {
-  const { open, setOpen, wrapperRef, multiselect } =
+  const { open, setOpen, setSelected, wrapperRef, multiselect } =
     React.useContext(ComboboxContext);
+
+  useEffect(() => {
+    setSelected(props.selected);
+  }, [props.selected]);
```

Comment:
hmmm I'm trying to think of a way to avoid making another copy of `props.selected` here

I worry this is getting pretty confusing where if it's `onSelect` you are operating on `props.selected` which is now `selected` from the provider

but also we are operating on `props.selected` via `selectedOptions` which already figures out for you if we are either using `props.selected` or the internal state

I feel like there's gotta be a way we can leverage the props/state we already have to track this

---------

if all we're doing is making a copy of `props.selected` to put into the provider to make it available to the trigger and we only ever use `setSelected` once, I'm tempted to just pass that as a prop to the provider when we initialize it instead

Reply:
yeah, once we do the new API, we'll wack this out since we don't need to. I'd actually do that on my PR since I have access to the selected now and can pass it in

Reply:
haha I ended up doing the same thing in my PR too, just passing it in to the provider